src/plugins/mini/mini.c

```diff
@@ -0,0 +1,218 @@
+/**
+ * @file
+ *
+ * @brief Source for mini plugin
+ *
+ * @copyright BSD License (see doc/LICENSE.md or http://www.libelektra.org)
+ *
+ */
+
+/* -- Imports --------------------------------------------------------------------------------------------------------------------------- */
+
+#include "mini.h"
+
+#include <kdbease.h>
+#include <kdberrors.h>
+#include <kdbhelper.h>
+#include <kdblogger.h>
+#include <kdbutility.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+/* -- Functions ------------------------------------------------------------------------------------------------------------------------- */
+
+// ===========
+// = Private =
+// ===========
+
+static inline KeySet * elektraMiniContract ()
+{
+	return ksNew (30, keyNew ("system/elektra/modules/mini", KEY_VALUE, "mini plugin waits for your orders", KEY_END),
+		      keyNew ("system/elektra/modules/mini/exports", KEY_END),
+		      keyNew ("system/elektra/modules/mini/exports/get", KEY_FUNC, elektraMiniGet, KEY_END),
+		      keyNew ("system/elektra/modules/mini/exports/set", KEY_FUNC, elektraMiniSet, KEY_END),
+#include ELEKTRA_README (mini)
+		      keyNew ("system/elektra/modules/mini/infos/version", KEY_VALUE, PLUGINVERSION, KEY_END), KS_END);
+}
+
+static inline char * stripComment (char * line)
+{
+	char * current = line;
+	char * before = NULL;
+
+	/* As long as we are not the end of the string and
+	   the current character is either not a comment marker or the comment marker was escaped */
+	while (*current != '\0' && ((*current != '#' && *current != ';') || (before && *before == '\\')))
+	{
+		before = current;
+		current++;
+	}
+	*current = '\0';
+	return line;
+}
+
+static inline char * findUnescapedEquals (char * text)
+{
+	char * equals = text;
+	char * before = NULL;
+
+	while (*equals != '\0' && (*equals != '=' || (before && *before == '\\')))
+	{
+		before = equals++;
+	}
+	return equals;
+}
+
+static inline char * unescape (char const * const text)
+{
+	return elektraReplace (text, "\\=", "=");
+}
+
+static inline void parseLine (char * line, size_t lineNumber, KeySet * keySet, Key * parentKey)
+{
+	char * pair = elektraStrip (stripComment (line));
+
+	if (*pair == '\0')
+	{
+		return;
+	}
+
+	char * equals = findUnescapedEquals (pair);
+	if (*equals == '\0' || equals == pair)
+	{
+		fprintf (stderr, "Ignored line %lu since “%s” does not contain a valid key value pair\n", lineNumber, pair);
+		ELEKTRA_ADD_WARNINGF (175, parentKey, "Line %lu: “%s” is not a valid key value pair", lineNumber, pair);
+		return;
+	}
+
+	*equals = '\0';
+
+	char * name = unescape (elektraRstrip (pair, NULL));
+	char * value = unescape (elektraLskip (equals + 1));
+
+	Key * key = keyNew (keyName (parentKey), KEY_END);
+	keyAddName (key, name);
+	keySetString (key, value);
+	ELEKTRA_LOG_DEBUG ("Name:  “%s”", keyName (key));
```

Comment:
I think that mainly depends on the preference of the user and the length of key and value. If both are very long, then the current solution should be better. If they are very short, it might not make sense to split them into two lines. 

Anyway, I like it this way, so I will leave it as is for now. If you have a strong preference for a one line solution, then please leave a comment below and tell me why the current solution is horrible 😊.